### ISSUE TITLE
Bug in categorical_crossentropy loss (label smoothing)

### ISSUE THREAD

**Issue by praveenjune17** [tensorflow/tensorflow#32231]
Possible bug found in categorical_crossentropy's label smoothing argument in [line](https://github.com/tensorflow/tensorflow/blob/r1.14/tensorflow/python/keras/losses.py#L940-L941) and for TF2 its in [line](https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/losses.py#L960-L961)

The axis specified in num_class is 1 which works fine for a  Rank 2 tensor but fails for higher rank tensors.

def _smooth_labels():
    num_classes = math_ops.cast(array_ops.shape(y_true)[1], y_pred.dtype)   #  -1 instead of 1 
    return y_true * (1.0 - label_smoothing) + (label_smoothing / num_classes)

  y_true = smart_cond.smart_cond(label_smoothing,
                                 _smooth_labels, lambda: y_true)
  return K.categorical_crossentropy(y_true, y_pred, from_logits=from_logits)

**Comment by jvishnuvardhan** [comment]
@praveenjune17 Sorry for the delay in response. Could you provide a simple standalone code to reproduce the issue? Also, provide platform details. Thanks!

**Comment by praveenjune17** [comment]
@jvishnuvardhan Raised this [PR](https://github.com/tensorflow/tensorflow/pull/32274) and added a Colab notebook which explains the issue.
